buildbot_nix/repo_config/__init__.py

```diff
@@ -0,0 +1,40 @@
+import tomllib
+from tomllib import TOMLDecodeError
+from typing import TYPE_CHECKING, Self
+
+from buildbot.process.buildstep import ShellMixin
+
+if TYPE_CHECKING:
+    from buildbot.process.log import Log
+from pydantic import BaseModel, ValidationError
+
+
+class RepoConfig(BaseModel):
+    branches: list[str]
+
+
+class BranchConfig(BaseModel):
+    lock_file: str = "flake.lock"
```

Comment:
Also please add some motivation to our documentation when to use this feature.

Reply:
Is this feature useful if you can just specify one lock_file? I thought this was about having some sort of build matrix?

Reply:
So you can have a nixpkgs-unstable flake.lock variant and a nixpkgs stable flake.lock variant and you test your flake against both without having to do all the plumbing in the flake. Or how do you envision to use this.

Reply:
my use case is i have a `flake-secret.lock` which has the input `secret` overriden to my "secret" flake for eval time secrets. Currently I do `--reference-lock-file` and `--output-lock-file` using a wrapper around Nix on my laptop. Ideally I'd have a `flake-ci.lock` or something.

Reply:
Ok. Please add this as an example somewhere. Maybe a seperate paragraph or somewhere close the table, that you added.

Reply:
ok

Reply:
done